java/src/jmri/jmrit/vsdecoder/VSDGeoFile.java

```diff
@@ -232,15 +232,8 @@ private void readGeoInfos() {
                     np = c1.getChildText("reporter-systemname");
                     Reporter rep = jmri.InstanceManager.getDefault(jmri.ReporterManager.class).getBySystemName(np);
                     if (rep != null) {
-                        String repNumber = np.substring(2); // connection prefix 3 signs?
-                        // An internal Reporter System Name can have non-numeric parts - do not allow here
-                        if (org.apache.commons.lang3.StringUtils.isNumeric(repNumber)) {
-                            rep_int = Integer.parseInt(repNumber);
-                            reporterlist[setup_index].add(rep_int);
-                        } else {
-                            log.warn("File {}: Reporter System Name {} is not valid for VSD", VSDGeoDataFileName, np);
-                            num_issues++;
-                        }
+                        rep_int = Integer.parseInt(jmri.Manager.getSystemSuffix(rep.getSystemName()));
+                        reporterlist[setup_index].add(rep_int);
```

Comment:
Thanks for doing this change.

Are you sure the system suffix doesn't have any non-numeric parts? If it does, this code will throw an exception.

A suggestion is to put these lines within a try-catch block and log a warning if an exception occurs. The method `parseInt()` throws a [NumberFormatException](https://docs.oracle.com/javase/8/docs/api/java/lang/NumberFormatException.html) if the string is not an integer.

https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html#parseInt-java.lang.String-

Reply:
Done. Thanks.